### PR TITLE
Refines the PR-template with the testability aspects.

### PR DESCRIPTION
**Pull request by ameteiko**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -1,19 +1,68 @@
...
+
+Thank you for your contribution!
+
+Before submitting your pull request, please follow these steps:
+
+1. Adhere to contributing guidelines, templates, the recommended Git workflow, and related documentation.
+2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/docs/contributing/02-contributing.md).
 3. Test your changes and attach their results to the pull request.
 4. Update the relevant documentation.
+5. Fill in the checklist below.
```
Comment (NHingerl) [inline .github/pull-request-template.md:11]: Below are three checklists: "Traceability", "Testability" (arguably, this one contains two lists, upping the total to four), and "Checklist". Which checklist do you mean here? Or all of them?

File: .github/pull-request-template.md [hunk 2]
```diff
@@ -1,19 +1,68 @@
...
+- [ ] Yes.
+- [ ] No, because e2e-tests are not needed.
+- [ ] No, because of ...
+
+<!--
+Please describe the tests you ran to verify your changes if needed. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
+-->
+Tests conducted for the PR:
+
+- Test description goes here
```
Comment (NHingerl) [inline .github/pull-request-template.md:58]:
```suggestion
- <!-- Test description goes here -->
```

File: .github/pull-request-template.md [hunk 3]
```diff
@@ -1,19 +1,68 @@
...
+
+## Checklist
+
+- [ ] My code follows the style guidelines of this project.
```
Comment (NHingerl) [inline .github/pull-request-template.md:62]: Can we link to those style guidelines?

Comment (ameteiko) [reply .github/pull-request-template.md:62]: Unfortunately, we have them as "tribal knowledge" for now, so there is no document to be linked. This item in a checklist is an instigation to formalise them.

**Review by NHingerl — APPROVED** [review]

**Pull request merged**